statistics/selectivity.go

```diff
@@ -283,6 +283,67 @@ func (coll *HistColl) Selectivity(ctx sessionctx.Context, exprs []expression.Exp
 			ret *= selectionFactor
 		}
 	}
+
+	// Now we try to cover those still not covered DNF conditions using independence assumption.
+	if mask > 0 {
```

Comment:
LGTM.
This approach is somewhat ad-hoc, but I have no idea to make it more general.
PTAL @eurekaka

Reply:
I think a recursive `Selectivity()` is more general if possible as I mentioned in the above comment.